lib/asset-manifest-inserter.js

```diff
@@ -0,0 +1,54 @@
+var BroccoliCachingWriter = require('broccoli-caching-writer');
+var path = require('path');
+var fs = require('fs-extra');
+var metaReplacer = require('./meta-handler').replacer;
+
+/**
+ * A Broccoli Plugin to modify index.html to include the asset manifest.
+ *
+ * @class AssetManifestInserter
+ * @extends BroccoliCachingWriter
+ *
+ * @private
+ * @param {Tree} tree
+ * @param {object} options
+ */
+
+AssetManifestInserter.prototype = Object.create(BroccoliCachingWriter.prototype);
+AssetManifestInserter.prototype.constructor = AssetManifestInserter;
+function AssetManifestInserter(inputNodes, options) {
+  options = options || {};
+  BroccoliCachingWriter.call(this, inputNodes, {
+    annotation: options.annotation
+  });
+  this.options = options;
+}
+
+AssetManifestInserter.prototype.build = function() {
+  var sourceDir = this.inputPaths[0];
+  var manifestFilePath = path.join(sourceDir, 'asset-manifest.json');
+  var indexFilePath = path.join(sourceDir, 'index.html');
+  var testIndexFilePath = path.join(sourceDir, 'tests', 'index.html');
+
+  var manifest;
+  try {
+    manifest = fs.readJsonSync(manifestFilePath);
+  } catch (error) {
+    console.warn('\n\nWarning: Unable to read asset-manifest.json from build with error: ' + error)
+    console.warn('Warning: Proceeding without generated manifest; you will need to manually provide a manifest to the Asset Loader Service to load bundles at runtime. If this was intentional you can turn this message off via the `noManifest` flag.\n\n');
+    manifest = { bundles: {} };
+  }
+
+  if (fs.existsSync(indexFilePath)) {
```

Comment:
`existsSync` is deprecated ([ref](https://nodejs.org/api/fs.html#fs_fs_existssync_path)), but we can use `ensureFileSync` from [fs-extra](https://www.npmjs.com/package/fs-extra#ensurefilefile-callback) instead.